### PR TITLE
Return a clearer error when terraform is used on an unsupported provider

### PR DESCRIPTION
**Pull request by rifelpet**
ref: #11951

How I determined the list of supported providers:

```
$ grep -rl RenderTerraform upup/pkg/fi/cloudup  | xargs -L1 dirname | uniq -c
   3 upup/pkg/fi/cloudup/spotinsttasks
  13 upup/pkg/fi/cloudup/gcetasks
  36 upup/pkg/fi/cloudup/awstasks
  16 upup/pkg/fi/cloudup/alitasks
   1 upup/pkg/fi/cloudup/dotasks
```

### CODE REVIEW

**Comment by johngmyers** [comment]
It would be good to extend `main.completeTarget()` to only include `TargetTerraform` when the provider supports it. (and the same for `TargetCloudformation`)

**Comment by rifelpet** [comment]
@johngmyers I added completion for `create cluster` but didn't touch `update cluster` because the cloud provider isn't in UpdateClusterOptions and I wasn't sure how to get access to the ClusterSpec from within the completion function.

**Comment by johngmyers** [comment]
@rifelpet You use `GetClusterForCompletion()` to get access to the cluster spec.

File: upup/pkg/fi/cloudup/apply_cluster.go [hunk 1]
```diff
@@ -89,6 +89,8 @@ var (
 	OldestSupportedKubernetesVersion = "1.17.0"
 	// OldestRecommendedKubernetesVersion is the oldest kubernetes version that is not deprecated in Kops
 	OldestRecommendedKubernetesVersion = "1.19.0"
+	// TerraformCloudProviders is the list of cloud providers with terraform target support
+	TerraformCloudProviders = []kops.CloudProviderID{kops.CloudProviderAWS, kops.CloudProviderGCE, kops.CloudProviderALI}
```
Comment (johngmyers) [inline upup/pkg/fi/cloudup/apply_cluster.go:93]: Could use a `sets.String()`

**Review by johngmyers — APPROVED** [review]

**Review by johngmyers — APPROVED** [review]

**Pull request merged**